packages/merge-styles/src/Stylesheet.ts

```diff
@@ -90,33 +98,57 @@ export interface ISerializedStylesheet {
   rules: Stylesheet['_rules'];
 }
 
-const STYLESHEET_SETTING = '__stylesheet__';
+export const STYLESHEET_SETTING = '__stylesheet__';
+
 /**
  * MSIE 11 doesn't cascade styles based on DOM ordering, but rather on the order that each style node
  * is created. As such, to maintain consistent priority, IE11 should reuse a single style node.
  */
 const REUSE_STYLE_NODE = typeof navigator !== 'undefined' && /rv:11.0/.test(navigator.userAgent);
 
-let _global: (Window | {}) & {
+declare global {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  interface Document {
+    adoptedStyleSheets: CSSStyleSheet[];
+  }
+}
+
+export type WindowWithMergeStyles = (Window | {}) & {
   [STYLESHEET_SETTING]?: Stylesheet;
+  [SHADOW_DOM_STYLESHEET_SETTING]?: typeof Stylesheet;
   FabricConfig?: {
     mergeStyles?: IStyleSheetConfig;
     serializedStylesheet?: ISerializedStylesheet;
   };
-} = {};
+};
+
+let _global: WindowWithMergeStyles = {};
 
 // Grab window.
 try {
   // Why the cast?
   // if compiled/type checked in same program with `@fluentui/font-icons-mdl2` which extends `Window` on global
   // ( check packages/font-icons-mdl2/src/index.ts ) the definitions don't match! Thus the need of this extra assertion
-  _global = (window || {}) as typeof _global;
+  _global = (window || {}) as WindowWithMergeStyles;
```

Comment:
this would also break SSR, maybe this is ensured to run only on client since we haven't had issues?
```suggestion
  _global = (typeof window !== 'undefined' ? window : {}) as WindowWithMergeStyles;
```